### PR TITLE
Feature/nex 939/new qti samples

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-778

Backport for the new QTI Test Samples ( needed for RC34 )

### CODE REVIEW

File: install/local/addQTIExamples.php [hunk 1]
```diff
@@ -15,18 +17,29 @@
...
-$file = dirname(__FILE__) . DIRECTORY_SEPARATOR . 'qtiv2p1Examples.zip';
+$testService = ServiceManager::getServiceManager()->get(taoQtiTest_models_classes_QtiTestService::class);
+$testClass = $testService->getRootClass();
 
-try {
-    $report = taoQtiTest_models_classes_QtiTestService::singleton()->importMultipleTests(new core_kernel_classes_Class(TaoOntology::TEST_CLASS_URI), $file);
-} catch (Exception $e) {
-    common_Logger::e('An error occured while importing QTI Test Example. The system reported the following error: ' . $e->getMessage());
-    throw $e;
+foreach (new DirectoryIterator(__DIR__) as $fileInfo) {
```
Comment (mike-ionut-mihai-sandu-tao) [inline install/local/addQTIExamples.php:30]: do we still need move this inside try-catch,
is it possible that the DirectoryIterator will fail ?

File: install/local/addQTIExamples.php [hunk 2]
```diff
@@ -15,18 +17,29 @@
...
+    if ($fileInfo->isReadable() && $fileInfo->isFile() && 'zip' === $fileInfo->getExtension()) {
+        try {
```
Comment (mike-ionut-mihai-sandu-tao) [inline install/local/addQTIExamples.php:32]: can we run psr-12

File: install/local/addQTIExamples.php [hunk 3]
```diff
@@ -15,18 +17,26 @@
...
+$testService = ServiceManager::getServiceManager()->get(taoQtiTest_models_classes_QtiTestService::class);
+$testClass = $testService->getRootClass();
+$samplesDirectory = new DirectoryIterator(__DIR__);
 
 try {
-    $report = taoQtiTest_models_classes_QtiTestService::singleton()->importMultipleTests(new core_kernel_classes_Class(TaoOntology::TEST_CLASS_URI), $file);
-} catch (Exception $e) {
-    common_Logger::e('An error occured while importing QTI Test Example. The system reported the following error: ' . $e->getMessage());
+    foreach ($samplesDirectory as $file) {
+        if ($file->isReadable() && $file->isFile() && 'zip' === $file->getExtension()) {
```
Comment (gabrielfs7) [inline install/local/addQTIExamples.php:32]: If the file is not readable we just bypass?

File: install/local/addQTIExamples.php [hunk 4]
```diff
@@ -15,18 +17,26 @@
...
+            $report = $testService->importMultipleTests($testClass, $file->getRealPath());
```
Comment (gabrielfs7) [inline install/local/addQTIExamples.php:33]: I am getting this error: 

```
Fatal error: Uncaught Error: Cannot pass parameter 7 by reference in /var/www/html/taoQtiTest/models/classes/class.QtiTestService.php:583
Stack trace:
#0 /var/www/html/taoQtiTest/models/classes/class.QtiTestService.php(372): taoQtiTest_models_classes_QtiTestService->importTest(Object(core_kernel_classes_Class), Object(taoQtiTest_models_classes_QtiResource), Object(taoQtiTest_models_classes_ManifestParser), '/tmp/tmp2013265...', Array)
#1 /var/www/html/taoQtiTest/install/local/addQTIExamples.php(33): taoQtiTest_models_classes_QtiTestService->importMultipleTests(Object(core_kernel_classes_Class), '/var/www/html/t...')
#2 /var/www/html/taoQtiTest/scripts/update/Updater.php(1963): require_once('/var/www/html/t...')
#3 /var/www/html/generis/common/ext/class.UpdateExtensions.php(118): oat\taoQtiTest\scripts\update\Updater->update('38.5.0')
#4 /var/www/html/generis/common/ext/class.UpdateExtensions.php(65): common_ext_UpdateExtensions->updateExtension(Object(common_ext_Extension))
#5 /var/www/html/tao/models/classes/extension/Up in /var/www/html/taoQtiTest/models/classes/class.QtiTestService.php on line 583
```

In `taoQtiTest_models_classes_QtiTestService` line 583 we must set the 7th parameter to a variable before.

Comment (gabrielfs7) [reply install/local/addQTIExamples.php:33]: I had incompatible version. Ignore it.

File: install/local/addQTIExamples.php (hunk 3 above)
Comment (mike-ionut-mihai-sandu-tao) [reply install/local/addQTIExamples.php:32]: we  are logging the info , as per before behavior

**Pull request merged**